### PR TITLE
Update step_time_event's behavior and code style to be more in line with recipe::step_holiday

### PR DESCRIPTION
**Pull request by joranE**
Took a pass at updating just this one step to run by you and get feedback. This pull request includes:

- The updated class name for the almanac objects, otherwise nothing works
- Switch to returning integer binary variables rather than boolean, to more closely align with `step_holiday` which I think these steps are most closely related to
- Adds the `keep_original_cols` argument with a default of `TRUE` as in `step_holiday`
- Refactors the `bake` method slightly, again to align with the approach in `step_holiday` and to replace `purrr` functions with `lapply`.
- Updates the tests to align with the modifications above

Basically, I just wanted to take a stab at a single function and get a sense of whether any of this is in the general direction you'd like to see.

### CODE REVIEW

File: R/time_event.R [hunk 1]
```diff
@@ -38,6 +42,7 @@ step_time_event <-
            trained = FALSE,
            rules = list(),
            columns = NULL,
+           keep_original_cols = TRUE,
```
Comment (EmilHvitfeldt) [inline R/time_event.R:45]: good job aligning with `step_holiday()`. I like you to set the default to `FALSE`. I don't like the default that `step_holiday()` has and would like to change it long term 😄

File: R/time_event.R [hunk 2]
```diff
@@ -99,32 +106,39 @@ prep.step_time_event <- function(x, training, info = NULL, ...) {
...
 #' @export
 bake.step_time_event <- function(object, new_data, ...) {
-  if (length(object$column) == 0L) {
-    # Empty selection
-    return(new_data)
-  }
+  col_names <- names(object$columns)
+  check_new_data(col_names, object, new_data)
+
+  for (col_name in col_names){
```
Comment (EmilHvitfeldt) [inline R/time_event.R:120]:
```suggestion
  for (col_name in col_names) {
```

File: R/time_event.R [hunk 3]
```diff
@@ -99,32 +106,39 @@ prep.step_time_event <- function(x, training, info = NULL, ...) {
...
+      rules = object$rules,
+      column = object$columns[col_name],
+      name = col_name,
+      new_data = new_data
+    )
 
-  new_cols <- purrr::imap_dfc(object$columns, time_event_helper,
-                              new_data, object$rules)
+    names(tmp) <- paste(col_name, names(tmp), sep = "_")
+    tmp[] <- lapply(X = tmp, FUN = vctrs::vec_cast, integer())
```
Comment (EmilHvitfeldt) [inline R/time_event.R:129]: What does the `[]` do after `tmp` here? is is needed?

File: R/time_event.R [hunk 4]
```diff
@@ -99,32 +106,39 @@ prep.step_time_event <- function(x, training, info = NULL, ...) {
...
+  new_data <- remove_original_cols(new_data, object, col_names)
   new_data
 }
 
-time_event_helper <- function(columnn, name, new_data, rule) {
-  res <- purrr::map_dfc(rule, ~ alma_in(new_data[[columnn]], .x))
-
-  names(res) <- paste(name, names(res), sep = "_")
+get_time_events <- function(rules, column, name, new_data) {
+  res <- lapply(X = rules, FUN = \(x) almanac::alma_in(new_data[[column]],x))
```
Comment (EmilHvitfeldt) [inline R/time_event.R:140]: We can't use the `\(x)` shorthand yet since it would lock out past versions of R. 

```suggestion
  res <- lapply(X = rules, FUN = function(x) almanac::alma_in(new_data[[column]], x))
```

**Review by EmilHvitfeldt — APPROVED** [review]
Looking good! I have a few notes with suggested edits!

File: R/time_event.R (hunk 3 above)
Comment (joranE) [reply R/time_event.R:129]: Without the `[]` the result of `lapply` will be a list and you'd have to do another `as_tibble` call. `df[] <- lapply(df,...)` is a pretty common base R pattern for iterating over columns and getting a data frame back, in my experience. If you think it's too cryptic I can just do

```
tmp <- lapply(X = tmp,...)
tmp <- as_tibble(tmp)
```

Happy to do whichever...

File: R/time_event.R (hunk 1 above)
Comment (joranE) [reply R/time_event.R:45]: Will do! Funnily enough the default of `FALSE` in `step_holiday` &`step_date` always sort of made sense to me.

File: R/time_event.R (hunk 3 above)
Comment (EmilHvitfeldt) [reply R/time_event.R:129]: looks good. Keep it!

**Comment by EmilHvitfeldt** [comment]
pushed a couple of commits to fix the CI check

**Comment by EmilHvitfeldt** [comment]
Everything looks good @joranE ! Thank you so much for you work!
